### DISCUSSION TITLE
Error in tls.lua

### DISCUSSION THREAD

**Discussion by NeoN7777** [daurnimator/lua-http#210] [Q&A]
Hello!
I'm trying to use lua-http on a pretty old device. When I run my lua script I get this error:
`luajit: /usr/local/share/lua/5.1/http/tls.lua:750: attempt to perform arithmetic on field 'OP_NO_COMPRESSION' (a nil value)`

As far as I understand, the problem is due to my version of openSSL - it is OpenSSL 0.9.8o 01 Jun 2010. It doesn't have SSL_OP_NO_COMPRESSION define, which is used here: https://github.com/wahern/luaossl/blob/8e9622c22f76d550594212c8bf5057f39d76f383/src/openssl.c#L10433

I'm using cross-compiled luaossl (20170903)
Is there any way to fix it?

**Comment by daurnimator** [comment]
That OpenSSL is too old to work on the modern internet. You'll need to upgrade your openssl version.
